runtime/base/src/xcm_config.rs

```diff
@@ -81,67 +76,44 @@ pub type LocationToAccountId = (
 	AccountId32Aliases<RelayNetwork, AccountId>,
 );
 
-/// Means for transacting the native currency on this chain.
-pub type LocalAssetTransactor = CurrencyAdapter<
+/// Means for transacting assets on this chain.
+pub type CurrencyTransactor = CurrencyAdapter<
 	// Use this currency:
 	Balances,
 	// Use this currency when it is a fungible asset matching the given location or name:
-	IsConcrete<SelfReserve>,
-	// Convert an XCM MultiLocation into a local account id:
+	IsConcrete<RelayLocation>,
+	// Convert an AccountId32 MultiLocation into a native chain account ID:
 	LocationToAccountId,
 	// Our chain's account ID type (we can't get away without mentioning it explicitly):
 	AccountId,
-	// We don't track any teleports of `Balances`.
+	// We don't track any teleports.
 	(),
 >;
 
 /// Means for transacting assets besides the native currency on this chain.
-pub type LocalFungiblesTransactor = FungiblesAdapter<
+pub type FungiblesTransactor = FungiblesAdapter<
 	// Use this fungibles implementation:
 	Assets,
 	// Use this currency when it is a fungible asset matching the given location or name:
 	ConvertedConcreteAssetId<
 		AssetId,
 		Balance,
-		AsPrefixedGeneralIndex<AssetsPalletLocation, AssetId, JustTry>,
-		JustTry,
-	>,
-	// Convert an XCM MultiLocation into a local account id:
-	LocationToAccountId,
-	// Our chain's account ID type (we can't get away without mentioning it explicitly):
-	AccountId,
-	// We don't track any teleports of `Assets`.
-	Nothing,
-	// We don't track any teleports of `Assets`.
-	CheckingAccount,
->;
-
-/// Means for transacting reserved fungible assets.
-/// AsAssetMultiLocation uses pallet_asset_registry to convert between AssetId and MultiLocation.
-pub type ReservedFungiblesTransactor = FungiblesAdapter<
-	// Use this fungibles implementation:
-	Assets,
-	// Use this currency when it is a registered fungible asset matching the given location or name
-	// Assets not found in AssetRegistry will not be used
-	ConvertedRegisteredAssetId<
-		AssetId,
-		Balance,
-		AsAssetMultiLocation<AssetId, AssetRegistry>,
+		AsPrefixedGeneralIndex<StatemineAssetsPalletLocation, AssetId, JustTry>,
 		JustTry,
 	>,
 	// Convert an XCM MultiLocation into a local account id:
 	LocationToAccountId,
 	// Our chain's account ID type (we can't get away without mentioning it explicitly):
 	AccountId,
-	// We don't track any teleports of `Assets`.
-	Nothing,
-	// We don't track any teleports of `Assets`.
+	// We only want to allow teleports of known assets. We use non-zero issuance as an indication
+	// that this asset is known.
+	NonZeroIssuance<AccountId, Assets>,
```

Comment:
do we have an integration test on teleporting assets?
it seems we've only tested teleports on native tokens so far... tbh I don't really have a mental model on how teleports would work on assets

Reply:
Not yet - took a quick look at manually teleporting xUSD from Statemine to Trappist/Base(3000) and both received the xcm message, but failed with:

```
xcmpQueue.Fail
error: XcmV2TraitsError
UntrustedTeleportLocation
```

I then tweaked the config of the `base-runtime` to match that of the reserve-transfer setup and seemed to be able to successfully teleport xUSD to the 'Base Local' chain.

```
impl xcm_executor::Config for XcmConfig {
	type IsReserve = Reserves;
-	type IsTeleporter = (); // Teleporting is disabled.
+	type IsTeleporter = Reserves;

impl pallet_xcm::Config for Runtime {
-	type XcmTeleportFilter = Nothing;
+	type XcmTeleportFilter = Everything;
	type XcmReserveTransferFilter = Everything;
```

![image](https://user-images.githubusercontent.com/60948618/202405347-5e6b00e8-edb0-4496-9c9c-ca0b5d108696.png)

Shall I update this PR with these changes, and then add a quick integration test to the other PR?

Reply:
Teleporting only makes sense between two different consensus systems (e.g. ETH <-> Polkadot) or between the relay chain and a common-good parachain (e.g. Polkadot -> Statemint), and there must exists a trust relationship (the sending chain trusts the receiving one to do proper accounting and not to mess up the token's emission / supply).

Reply:
So in the context of Trappist, the only Teleport use case we have is for teleporting the relay chain's (Rococo) native token to Statemine.

Reply:
Thanks for clarifying!

Reply:
I'll remove it 👍

Reply:
Should be good to go. 

Also fixed the build script (in line with GH workflow changes) after the recent PR to set trappist-runtime as default feature. It was a quick fix so didnt think it warranted a separate PR, but can revert if you'd prefer.